File: sdks/ts/src/services.ts
import { createPromiseClient, type Transport } from "@connectrpc/connect";
import { SyncService } from "./gen/flint/v1/sync_connect.js";
import { AgentService } from "./gen/flint/v1/agent_connect.js";
import { SignalService } from "./gen/flint/v1/signal_connect.js";
import { EntityService } from "./gen/flint/v1/entity_connect.js";
import { AuthzService } from "./gen/flint/v1/authz_connect.js";

/**
 * Thin typed clients for the gateway services beyond SpineService.
 *
 * All of these now have a live gateway server implementation: Sync (CRDT sync),
 * Agent (agent runs), Signal (WebRTC signaling), Entity (read/watch, live since
 * p17-c004), and Authz (relation check/write/delete, live since p17-c005).
 */

export const createSyncClient = (transport: Transport) =>
  createPromiseClient(SyncService, transport);

export const createAgentClient = (transport: Transport) =>
  createPromiseClient(AgentService, transport);

export const createSignalClient = (transport: Transport) =>
  createPromiseClient(SignalService, transport);

export const createEntityClient = (transport: Transport) =>
  createPromiseClient(EntityService, transport);

export const createAuthzClient = (transport: Transport) =>
  createPromiseClient(AuthzService, transport);

export { SyncService, AgentService, SignalService, EntityService, AuthzService };

File: sdks/go/client/services.go
package client

import (
	"net/http"

	"connectrpc.com/connect"

	"github.com/prometheusags/frf/sdks/go/gen/flint/v1/flintv1connect"
)

// Thin clients for the gateway services beyond SpineService.
//
// All of these now have a live gateway server implementation: Sync (CRDT sync),
// Agent (agent runs), Signal (WebRTC signaling), Entity (read/watch, live since
// p17-c004), and Authz (relation check/write/delete, live since p17-c005).

// NewSyncClient constructs a SyncService client targeting baseURL.
func NewSyncClient(baseURL string, opts ...connect.ClientOption) flintv1connect.SyncServiceClient {
	return flintv1connect.NewSyncServiceClient(http.DefaultClient, baseURL, opts...)
}

// NewAgentClient constructs an AgentService client targeting baseURL.
func NewAgentClient(baseURL string, opts ...connect.ClientOption) flintv1connect.AgentServiceClient {
	return flintv1connect.NewAgentServiceClient(http.DefaultClient, baseURL, opts...)
}

// NewSignalClient constructs a SignalService client targeting baseURL.
func NewSignalClient(baseURL string, opts ...connect.ClientOption) flintv1connect.SignalServiceClient {
	return flintv1connect.NewSignalServiceClient(http.DefaultClient, baseURL, opts...)
}

// NewEntityClient constructs an EntityService client targeting baseURL.
func NewEntityClient(baseURL string, opts ...connect.ClientOption) flintv1connect.EntityServiceClient {
	return flintv1connect.NewEntityServiceClient(http.DefaultClient, baseURL, opts...)
}

// NewAuthzClient constructs an AuthzService client targeting baseURL.
func NewAuthzClient(baseURL string, opts ...connect.ClientOption) flintv1connect.AuthzServiceClient {
	return flintv1connect.NewAuthzServiceClient(http.DefaultClient, baseURL, opts...)
}
